### PR TITLE
client, db, server: Don't delete post if already deleted

### LINKED ISSUE #676: Don't log doubly deleted posts into modlog
Delete by IP produces  a lot of duplicates.

### PR DESCRIPTION
**Pull request by Chiiruno**
Fixes #676

### CODE REVIEW

**Comment by bakape** [comment]
What? No, no, no. I meant you should not be able to delete already deleted posts. It should do nothing.

**Comment by Chiiruno** [comment]
Oh, that's easy enough, I'll do that instead.

File: go/src/meguca/db/init.go [hunk 1]
```diff
@@ -605,6 +605,13 @@ var upgrades = []func(*sql.Tx) error{
 			`drop table pyu_limit`,
 		)
 	},
+	func(tx *sql.Tx) (err error) {
+		_, err = tx.Exec(
+			`update posts
+				set deleted = false`,
```
Comment (bakape) [inline go/src/meguca/db/init.go]: Not needed.

File: go/src/meguca/db/posts.go [hunk 2]
```diff
@@ -207,3 +207,9 @@ func SetPostCounter(c uint64) error {
 	_, err := db.Exec(`SELECT setval('post_id', $1)`, c)
 	return err
 }
+
+// GetPostDeleted retrieves a post's deletion status
+func GetPostDeleted(id uint64) (del bool, err error) {
+	err = prepared["get_post_deleted"].QueryRow(id).Scan(&del)
```
Comment (bakape) [inline go/src/meguca/db/posts.go]: Use sql.NullBool

File: go/src/meguca/db/posts.go [hunk 3]
```diff
@@ -207,3 +207,9 @@ func SetPostCounter(c uint64) error {
 	_, err := db.Exec(`SELECT setval('post_id', $1)`, c)
 	return err
 }
+
+// GetPostDeleted retrieves a post's deletion status
+func GetPostDeleted(id uint64) (del bool, err error) {
```
Comment (bakape) [inline go/src/meguca/db/posts.go]: No need to make this function public.

File: go/src/meguca/db/sql/init/init.sql [hunk 4]
```diff
@@ -124,7 +124,7 @@ create index sticky on threads (sticky);
 create table posts (
 	editing boolean not null,
 	spoiler boolean,
-	deleted boolean,
+	deleted boolean not null,
```
Comment (bakape) [inline go/src/meguca/db/sql/init/init.sql]: Not needed.

**Pull request merged**